cwl_runner/cwl_runner.sh

```diff
@@ -0,0 +1,223 @@
+  Show help message and exit
+
+Common options:
+-w --workflow-file PATH
+  REQUIRED. The absolute path to the .cwl workflow definition file in Cloud Storage.
+-s --settings-file PATH
+  REQUIRED. The absolute path to the .json settings file in Cloud Storage.
+-i --input GCS_PATH1[,GCS_PATH2,...]
+  The absolute path(s) in Cloud Storage to the file(s) that must be copied to the VM's local disk. 
+-I --input-recursive GCS_PATH1[,GCS_PATH2,...]
+  The absolute path(s) in Cloud Storage to the folder(s) that must be copied to the VM's local disk. 
+-m --machine-type STRING
+  The Google Compute Engine VM machine type name. Default: ${MACHINE_TYPE}.
+-o --output GCS_PATH
+  REQUIRED. The path where CWL outputs and logs will be copied after the workflow completes.
+
+Other options:
+-d --disk-size INT
+  The disk size in Gb. Default: ${DISK_SIZE}.
+-k --keep-alive
+  Leave the VM running after the workflow completes or fails so that you can ssh in for debugging.
+-p --preemptible
+  Run with a preemptible VM that costs less but may be terminated before finishing.
+-r --runner STRING
+  The CWL runner to use. Values can be "cwltool" or "rabix". Default: ${RUNNER}.
+-z --zone STRING
+  The zone to launch the VM and disk in. If omitted, your default project zone will be used.
+
+EOM
+
+# Parse command-line
+while [[ $# -gt 0 ]] 
+do
+key="$1"
+
+case $key in
+  -h|--help)
+  echo "${HELP_MESSAGE}"
+  exit 1
+  ;;
+  -w|--workflow-file)
+  WORKFLOW_FILE="$2"
+  shift
+  ;;
+  -s|--settings-file)
+  SETTINGS_FILE="$2"
+  shift
+  ;;
+  -i|--input)
+  INPUT="$2"
+  shift
+  ;;
+  -r|--input-recursive)
+  INPUT_RECURSIVE="$2"
+  shift
+  ;;
+  -o|--output)
+  OUTPUT="$2"
+  shift
+  ;;
+  -d|--disk-size)
+  DISK_SIZE="$2"
+  shift
+  ;;
+  -k|--keep-alive)
+  KEEP_ALIVE="true"
+  ;;
+  -m|--machine-type)
+  MACHINE_TYPE="$2"
+  shift
+  ;;
+  -p|--preemptible)
+  PREEMPTIBLE="--preemptible"
+  ;;
+  -z|--zone)
+  ZONE="--zone $2"
+  shift
+  ;;
+  -r|--runner)
+  RUNNER="$2"
+  shift
+  ;;
+  *)
+  # unknown option
+  ;;
+esac
+shift
+done
+
+set -e errexit
+
+if [[ -z "${WORKFLOW_FILE}" || -z "${SETTINGS_FILE}" || -z "${OUTPUT}" ]]; then
+  echo "Error: Missing required argument(s)."
+  echo "${HELP_MESSAGE}"
+  exit 1
+fi
+
+DISK_NAME="cwl-disk-${OPERATION_ID}"
+DISK_CMD="gcloud compute disks create ${DISK_NAME} ${ZONE} --size ${DISK_SIZE}"
+
+PWD="$( cd $( dirname ${BASH_SOURCE[0]} ) && pwd )"
+
+STARTUP_SCRIPT_NAME="cwl_startup.sh"
+STARTUP_SCRIPT="${PWD}/${STARTUP_SCRIPT_NAME}"
+STARTUP_SCRIPT_URL="${OUTPUT}/${STARTUP_SCRIPT_NAME%.*}-${OPERATION_ID}.sh"
+
+SHUTDOWN_SCRIPT_NAME="cwl_shutdown.sh"
+SHUTDOWN_SCRIPT="${PWD}/${SHUTDOWN_SCRIPT_NAME}"
+SHUTDOWN_SCRIPT_URL="${OUTPUT}/${SHUTDOWN_SCRIPT_NAME%.*}-${OPERATION_ID}.sh"
+
+STATUS_FILE="${OUTPUT}/status-${OPERATION_ID}.txt"
+STATUS="STARTING"
+
+VM_NAME="cwl-vm-${OPERATION_ID}"
+VM_CMD="gcloud compute instances create ${VM_NAME} \
+--disk name=${DISK_NAME},device-name=${DISK_NAME},auto-delete=yes \
```

Comment:
Indent all of the command flags.
It inserts spaces into the string, but it does not matter and reads better.